utils/utils.h

```diff
@@ -276,12 +276,14 @@ extern struct gbConf gbConf;
             snprintf(tmp, 1024, "%s/%s", vol?vol:"", blk->block_name); \
             if (fp) {                                                  \
               size_t newLen = fread(sr->out, sizeof(char), 8192, fp);  \
-              if (ferror( fp ) != 0)                                   \
+              if (ferror( fp ) != 0) {                                 \
                 LOG("mgmt", GB_LOG_ERROR,                              \
                     "reading command %s output for %s failed(%s)", tmp,\
                     cmd, strerror(errno));                             \
-              else                                                     \
+                sr->out[0] = '\0';                                     \
+              } else {                                                 \
                 sr->out[newLen++] = '\0';                              \
+              }                                                        \
               sr->exit = blockValidateCommandOutput(sr->out, opt,      \
                                                     (void*)blk);       \
```

Comment:
Here if fread fails, does it still need to call blockValidateCommandOutput() ?

```
if (ferror( fp ) != 0) {
  [...]
} else {
  sr->out[newLen++] = '\0';
  sr->exit = blockValidateCommandOutput(sr->out, opt,
  [...]
}
```

Reply:
+1 Xiubo

just set sr->exit = -1; and move on.